### PR TITLE
Update tests to use new roslyn and metadata reader

### PR DESCRIPTION
**Pull request by twsouthwick**
This also include tests that were skipped before due to needing a newer System.Reflection.Metadata.  Since this is available now, the skip attribute is removed.

### CODE REVIEW

File: tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs [hunk 1]
```diff
@@ -113,10 +115,15 @@ private void CompareSpecificDependency(string path, string v)
...
 
-            var found = dependencies.Dependencies
-                .Any(d => string.Equals(d.Key.MemberDocId, v, StringComparison.Ordinal));
+            foreach (var dependency in dependencies.Dependencies)
+            {
+                if (string.Equals(dependency.Key.MemberDocId, v, StringComparison.Ordinal))
+                {
+                    return;
+                }
+            }
```
Comment (mjrousos) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: Why the change here? Just a perf optimization?

File: tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs [hunk 2]
```diff
@@ -137,7 +144,16 @@ private void CompareDependencies(string path, IEnumerable<Tuple<string, int>> ex
...
+            Assert.Equal(expectedOrdered.Count, foundDocIds.Count);
+
+            foreach (var combined in expectedOrdered.Zip(foundDocIds, Tuple.Create))
+            {
+                var expectedItem = combined.Item1;
+                var actualItem = combined.Item2;
+
+                Assert.Equal(expectedItem.Item1, actualItem.Item1);
+                Assert.Equal(expectedItem.Item2, actualItem.Item2);
+            }
```
Comment (mjrousos) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: I'd have hoped that Assert.Equal would have done the same thing as this code already. I take it that isn't the case.

**Comment by conniey** [comment]
LGTM

**Comment by mjrousos** [comment]
A couple comments regarding things I'm curious about, but LGTM.

File: tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs (hunk 1 above)
Comment (twsouthwick) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: I think I did it for debugging purposes - I kept changing it to this to identify what the dependencies are.  FIgured it would be better to keep it.

File: tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs (hunk 2 above)
Comment (twsouthwick) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: Again, this is to help with debugging when something fails - I found that the Assert.Equal made it difficult to test to find what the failing issue was.

Comment (mjrousos) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: Ah, I see. Makes sense.

File: tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs (hunk 1 above)
Comment (mjrousos) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: I see.

**Comment by mjrousos** [comment]
LGTM
